R/g.analyse.perday.R

```diff
@@ -320,23 +324,58 @@ g.analyse.perday = function(ndays, firstmidnighti, time, nfeatures,
             for (mi in 2:ncol(metashort)) { #run through metrics (for features based on single metrics)
               NRV = length(which(is.na(as.numeric(as.matrix(vari[,mi]))) == FALSE))
               varnum = as.numeric(as.matrix(vari[,mi])) #varnum is one column of vari
+              vari2 = vari #vari2 is used in identify_levels when AskedToWear247 = FALSE, need to redefine here in every loop to account for qwindows
+              # indices to focus on for time-use variables (ilevels, recorded time, MVPA, ...)
+              recorded = 1:length(varnum) # this is redefined for first and last day
+              if (isTRUE(params_cleaning[["part2ExcludeNonwear"]])) {
+                varnum[which(val != 0)] = NA
+              }
               # # if this is the first or last day and it has more than includedaycrit number of days then expand it
               # Comment out the following 10 lines if you want to include only the actual data
-              if (NRV != length(averageday[, (mi - 1)])) {
+              if (NRV < length(averageday[, (mi - 1)])) { # modified to avoid imputing time in the daylight saving days (25-h long, then they meet the condition NRV != nrow(averageday))
                 difference = NRV - length(averageday[, (mi - 1)])
                 if (di == 1) {
                   varnum = c(averageday[1:abs(difference), (mi - 1)], varnum)
+                  # indices to focus on for time-use variables (ilevels, recorded time, MVPA, ...)
+                  if (abs(difference) > 0) recorded = abs(difference):length(varnum)
+                  # varnum = c(rep(NA, abs(difference)), varnum) # 2022-11-1: if day <24h, then the total time analised is also <24h
                   # readjust anwi indices in case that varnum has been imputed
                   if (max(anwi_t1) < length(varnum)) { # since GGIR always calculates full window, max(anwi_t1) should always equals length(varnum)
                     anwi_t0 = anwi_t0 + abs(difference)
                     anwi_t1 = anwi_t1 + abs(difference)
                     # then, we reset the minimum anwi_t0 to 1 to consider the imputed varnum
                     anwi_t0[which(anwi_t0 == min(anwi_t0))] = 1
                   }
+                  # impute vari timestamp as well
```

Comment:
I do not understand what these new 11 lines of code do. The comment `# impute vari timestamp as well` gives the impression that object `vari` is altered, but that is not the case. Can you please use more code comments in this section to clarify each step and please try to replace object names by a name that clarifies their content or purpose to improve readability.

I have also been working on this bit of code in relation to #728, so I want to make sure I fully understand what your version of the code tries to do to ease merging my code modifications when this branch is merged.

Reply:
I expanded the comment with:
```
# Readjust also the vari timestamp in the first day.
# In default settings this is not used and has not any effect on the output.
# If device is not worn during sleep (AskedToWear247 == FALSE),
# then, the vari matrix is used to derive the behavioral classes
# with identify_levels(), and there the timestamp needs to be readjusted.
# This is done in a new object (vari2), so that vari is never modified.
```